crates/indexer/src/runtime.rs

```diff
@@ -0,0 +1,208 @@
+            Arc::new(DefaultTransactionProcessor::new(conn_pool.clone()))
+        }
+        Processor::TokenProcessor => Arc::new(TokenTransactionProcessor::new(
+            conn_pool.clone(),
+            config.index_token_uri_data,
+        )),
+    };
+
+    let options = TransactionFetcherOptions::new(
+        None,
+        None,
+        Some(config.batch_size),
+        None,
+        config.fetch_tasks as usize,
+    );
+
+    let tailer = Tailer::new(context, conn_pool.clone(), processor, options)
+        .expect("Failed to instantiate tailer");
+
+    if !config.skip_migrations {
+        info!(processor_name = processor_name, "Running migrations...");
+        tailer.run_migrations();
+    }
+
+    let start_version = match config.starting_version {
+        None => tailer
+            .get_start_version(&processor_name)
+            .unwrap_or_else(|| {
+                info!(
+                    processor_name = processor_name,
+                    "Could not fetch version from db so starting from version 0"
+                );
+                0
+            }),
+        Some(version) => version,
+    };
+
+    info!(
+        processor_name = processor_name,
+        start_version = start_version,
+        "Setting starting version..."
+    );
+    tailer.set_fetcher_version(start_version).await;
+
+    info!(processor_name = processor_name, "Starting fetcher...");
+    tailer.transaction_fetcher.lock().await.start().await;
+
+    let start = chrono::Utc::now().naive_utc();
+
+    info!(
+        processor_name = processor_name,
+        start_version = start_version,
+        "Indexing loop started!"
+    );
+
+    let mut versions_processed: u64 = 0;
+    let mut total_processed: u64 = 0;
+    let mut base: u64 = 0;
+    let mut version_to_check_chain_id: u64 = 0;
+
+    // Check once here to avoid a boolean check every iteration
+    if config.check_chain_id {
+        tailer
+            .check_or_update_chain_id()
+            .await
+            .expect("Failed to get chain ID");
+        version_to_check_chain_id = versions_processed + 100_000;
+    }
+
+    let (tx, mut receiver) = tokio::sync::mpsc::channel(100);
+    let mut tasks = vec![];
+    for _ in 0..config.processor_tasks {
+        let other_tx = tx.clone();
+        let other_tailer = tailer.clone();
+        let task = tokio::task::spawn(async move {
+            loop {
+                let (num_res, res) = other_tailer.process_next_batch().await;
+                other_tx.send((num_res, res)).await.unwrap();
+            }
+        });
+        tasks.push(task);
+    }
+
+    let emit_every = config.emit_every as u64;
+
+    loop {
+        if config.check_chain_id && version_to_check_chain_id < versions_processed {
+            tailer
+                .check_or_update_chain_id()
+                .await
+                .expect("Failed to get chain ID");
+            version_to_check_chain_id = versions_processed + 100_000;
+        }
```

Comment:
Given this setup we don't need this since there's no way that we can switch the node without restarting the indexer. 

Another thing I want to check upon restart is whether max version in db > ledger version. If so we should panic and let user know that they should wipe the db b/c chain most likely reset. I can add this later tho.

Reply:
you can indeed- I've done it locally hahaha